holoviews/core/data/xarray.py

```diff
@@ -37,6 +37,22 @@ def dtype(cls, dataset, dim):
         return dataset.data[name].dtype
 
 
+    @classmethod
+    def shape(cls, dataset, gridded=False):
```

Comment:
This is fine if you are sure that the ``GridInterface`` implementation needs to be overridden. If not, it might be more correct to fix things in a more general way there.

Reply:
I was remembering things incorrectly when I mentioned this, the ``GridInterface`` enforces that the array dimensions must match the shape as defined as ``tuple(len(element.dimension_values(kd)) for kd in element.kdims)[::-1]`` so it does not allow transposing the dimensions without explicitly transposing  the data.